src/TtcVault.sol

```diff
@@ -508,49 +512,38 @@ contract TtcVault is ITtcVault, ReentrancyGuard {
-     * @param constituentTokenIndex The index of the token in the constituentTokens array
+     * @param tokenAddress The address of the token to get the price of
      * @return The number of ETH that has to be paid for 1 constituent token
      */
-    function getLatestPriceInEthOf(uint8 constituentTokenIndex) public view returns (uint256) {
-        address tokenAddress = constituentTokens[constituentTokenIndex].tokenAddress;
+    function getLatestPriceInEthOf(address tokenAddress) public view returns (uint256) {
         address wEthAddress = address(i_wEthToken);
 
         // if token is rETH (0 index), use native contract for better price accuracy
-        if (constituentTokenIndex == 0) {
+        if (tokenAddress == address(i_rEthToken)) {
             return i_rEthToken.getEthValue(1e18);
         }
 
         // get a token/wETH pool's address
         
-        // payload to get a pool address
-        bytes memory payload = abi.encodeWithSignature("getPool(address,address,uint24)", tokenAddress, wEthAddress, UNISWAP_PRIMARY_POOL_FEE);
-        (bool success, bytes memory data) = i_uniswapFactory.staticcall(payload);
-
-        // try to find this pool in all three fee tiers
-        // TODO: refactor this abomination, maybe provide fee tier as a param
-        if (!success) {
-            payload = abi.encodeWithSignature("getPool(address,address,uint24)", tokenAddress, wEthAddress, UNISWAP_SECONDARY_POOL_FEE);
-            (success, data) = i_uniswapFactory.staticcall(payload);
-            if (!success) {
-                payload = abi.encodeWithSignature("getPool(address,address,uint24)", tokenAddress, wEthAddress, UNISWAP_TERTIARY_POOL_FEE);
-                (success, data) = i_uniswapFactory.staticcall(payload);
-                if (!success) {
-                    payload = abi.encodeWithSignature("getPool(address,address,uint24)", tokenAddress, wEthAddress, UNISWAP_QUATERNARY_POOL_FEE);
-                    (success, data) = i_uniswapFactory.staticcall(payload);
-                    if (!success) {
-                        revert PoolDoesNotExist();
-                    }
-                }
-            }
-        }
-
-        address pool = abi.decode(data, (address));
+        address pool = getPoolWithFee(tokenAddress, wEthAddress, UNISWAP_PRIMARY_POOL_FEE);
         if (pool == address(0)) {
             revert PoolDoesNotExist();
         }
 
         // convert to IUniswapV3PoolState to get access to sqrtPriceX96
         IUniswapV3PoolState _pool = IUniswapV3PoolState(pool);
+        
+        uint24[3] memory feeTiers = [UNISWAP_SECONDARY_POOL_FEE, UNISWAP_TERTIARY_POOL_FEE, UNISWAP_QUATERNARY_POOL_FEE];
+
+        for (uint8 i; i < 3; i++) {
+            pool = getPoolWithFee(tokenAddress, wEthAddress, feeTiers[i]);
+            if (pool == address(0)) {
+                continue;
+            } else {
+                _pool = IUniswapV3PoolState(pool);
+                break;
+            }
+        }
```

Comment:
this is a refactored stuff that was before, looks better imo

Reply:
Agreed